File: lvl3/efficacite_temporelle/premier_absent.py
''''
    Name : Premier absent
    Author : Arthur G
    Date : 18 Aug 2021
'''
import sys
if __name__ == '__main__':
    n,p=map(int,sys.stdin.readline().split())
    data=[False]*(250001)
    min=0
    for i in range(p):
        eleve = int(sys.stdin.readline())
        eleve-=1
        if eleve < p:
            data[eleve]=True
        while min < n and data[min]:
            min+=1
        if min == n:
            sys.stdout.write("-1"+"\n")
        else:
            sys.stdout.write(str(min+1)+"\n")

File: lvl3/efficacite_temporelle/premier_absent.cpp
/*
    Name : Premier absent
    Author : Arthur G
    Date : 18 Aug 2021
*/

#include<bits/stdc++.h>

using namespace std;

int main()
{
    set<int> data;
    int P, N, eleve, min = 1;
    scanf("%d %d", &N, &P);
    for (int i = 1; i <= P; i++)
    {
        scanf("%d", &eleve);
        data.insert(eleve);
        if (i == N)
            printf("-1\n");
        else
        {
            if (min == eleve)
                for (set<int>::iterator it1 = data.begin(); it1 != data.end(); it1++)
                {
                    if (min != *it1)
                        break;
                    data.erase(min);
                    min++;
                }
            printf("%d\n", min);
        }
    }
    return 0;
}
